FreeRTOS_IP.c

```diff
@@ -2646,6 +2646,15 @@ static eFrameProcessingResult_t prvAllowIPPacketIPv4( const IPPacket_t * const p
                 /* Packet is not for this node, release it */
                 eReturn = eReleaseBuffer;
             }
+            else if( ( memcmp( ( void * ) xBroadcastMACAddress.ucBytes,
+                               ( void * ) ( pxIPPacket->xEthernetHeader.xDestinationAddress.ucBytes ),
+                               sizeof( MACAddress_t ) ) == 0 ) &&
+                     ( ( ulSourceIPAddress & 0xff ) != 0xff ) )
```

Comment:
Here `ulSourceIPAddress` is in network byte order, which means that you should apply ntohl:
~~~c
    ( ( FreeRTOS_ntohl( ulSourceIPAddress ) & 0xff ) != 0xff ) )
~~~
I will check first this to be sure.

Reply:
I think you wanted to test the destination address in stead of `ulSourceIPAddress`.
The line below works fine for me:
~~~c
    ( ( FreeRTOS_ntohl( ulDestinationIPAddress ) & 0xff ) != 0xff ) )
~~~
I tested this on a little-endian platform.